### PR TITLE
refactor(controller): fix slice init length

### PR DESCRIPTION
**Pull request by cuishuang**
The intention here should be to initialize a slice with a capacity of length rather than initializing the length of this slice. 

The online demo: https://go.dev/play/p/q1BcVCmvidW

### CODE REVIEW

File: internal/argocd/health.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func (h *applicationHealth) GetApplicationHealth(
 	// considered unhealthy as they may indicate a problem which can result in
 	// e.g. the health status result to become unreliable.
 	if errConditions := filterAppConditions(app, healthErrorConditions...); len(errConditions) > 0 {
-		issues := make([]error, len(errConditions))
+		issues := make([]error, 0, len(errConditions))
```
Comment (krancour) [inline internal/argocd/health.go:173]: Nice catch. But it's not _this_ line that's wrong. It's what comes after it. This is what it should have been:

```golang
		issues := make([]error, len(errConditions))
		for i, condition := range errConditions {
			issues[i] = fmt.Errorf(
				"Argo CD Application %q in namespace %q has %q condition: %s",
				appKey.Name,
				appKey.Namespace,
				condition.Type,
				condition.Message,
			)
		}
```

Comment (krancour) [reply internal/argocd/health.go:173]: Similar for the other spot.

Comment (cuishuang) [reply internal/argocd/health.go]: Modified.

Please review again.

**Comment by krancour** [comment]
Thank you @cuishuang!

**Review by krancour — APPROVED** [review]

**Pull request merged**